src/main/prepared_statement_data.cpp

```diff
@@ -13,23 +13,25 @@ PreparedStatementData::~PreparedStatementData() {
 
 void PreparedStatementData::Bind(vector<Value> values) {
 	// set parameters
-	if (values.size() != value_map.size()) {
+	const auto required = unbound_statement ? unbound_statement->n_param : 0;
```

Comment:
Perhaps this should be an internal exception if there is no unbound_statement?

Reply:
It came up at some point, which is why I added the check.

Reply:
I don't think it is an error, just a pass. Although it does remind me a bit of #3029.